pkg/render/apiserver.go

```diff
@@ -1040,6 +1037,14 @@ func (c *apiServerComponent) apiServerContainer() corev1.Container {
 			TimeoutSeconds:      10,
 		},
 	}
+	// In case of OpenShift, apiserver needs privileged access to write audit logs to host path volume.
+	// Audit logs are owned by root on hosts so we need to be root user and group. Audit logs are supported only in Enterprise version.
+	if c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise &&
+		c.cfg.Openshift {
```

Comment:
I think we need the RootContext for all `Variant == Enterprise`, so the `c.cfg.Openshift` should not be part of the if statement.

Reply:
Okay, I can make that change. 

So today, the `NewRootContext()` function sets the `RunAsGroup` and `RunAsUser` to `0`, and if it is Openshift, then it sets also `AllowPrivilegeEscalation` and `Privileged` to `true` value.

The Dockerfile defined RunAsGroup and RunAsUser values would be set if we would use the `NewNonRootContext()` function.

My goal here is to run the calico-apiserver as a non-root container and we don't use tigera-enterprise at the moment.

Thank you,
Adam

Reply:
Am I reading this if condition correct? This condition says that we will use `NewRootContext` only when Calico Enterprise *and* on openshift. It will break on all non-openshift Calico Enterprise cluster. I think we need to remove `&& c.cfg.Openshift` from the if condition.

```
c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise &&
		c.cfg.Openshift
```

Reply:
Looks good. Thanks for changing that.